crates/wasi-common/wig/src/wasi.rs

```diff
@@ -214,11 +214,15 @@ pub fn define_struct(args: TokenStream) -> TokenStream {
                                     #handle_early_error
                                 }
                             };
-                            hostcalls::#name_ident(
+                            let result = hostcalls::#name_ident(
                                 &mut my_cx.borrow_mut(),
                                 memory.data_unchecked_mut(),
                                 #(#hostcall_args),*
-                            ) #cvt_ret
+                            );
+                            match result {
+                                Ok(r) => {return Ok(r #cvt_ret);},
+                                Err(err) => { return Err(wasmtime::Trap::new(err)); },
+                            }
```

Comment:
FWIW this could also be:

```rust
match result {
    Ok(r) => Ok(r #cvt_ret),
    Err(err) => Err(wasmtime::Trap::new(err)),
}
```

Reply:
Truth.  Addressed in https://github.com/bytecodealliance/wasmtime/pull/2419/commits/6c717bf86cf7584aeaac694c3868298a0518c519